### ISSUE TITLE
--disable-openmp does not work

### ISSUE THREAD

**Issue by ClementPernet** [linbox-team/linbox#24]
If --disable-openmp is passed to the configure script, the compilation of the lib still adds a -fopenmp flag, thus generating warnings like:
```
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../.. -O2 -Wall -g -DNDEBUG -U_LB_DEBUG -DDISABLE_COMMENTATOR -fopenmp -I/home/pernet/Logiciels/install/install/include -D__FFLASFFPACK_HAVE_CBLAS -msse4.1 -mavx -fabi-version=6 -I/home/pernet/Logiciels/install/install/include -I/home/pernet/Logiciels/install/install/include -DFFLAS_COMPILED -DFFPACK_COMPILED -std=gnu++11 -c diophantine-solver.C  -fPIC -DPIC -o .libs/diophantine-solver.o
In file included from ../../linbox/linbox-config.h:30:0,
                 from diophantine-solver.C:25:
/home/pernet/Logiciels/install/install/include/fflas-ffpack/fflas-ffpack-config.h:118:2: warning: #warning "openmp was not detected correctly at configure time, please report this bug" [-Wcpp]
 #warning "openmp was not detected correctly at configure time, please report this bug"
```

**Comment by jdemeyer** [comment]
In `configure`, you have both `$with_omp` and `$with_openmp` but they should be the same variable.

**Comment by kiwifb** [comment]
Line 39 of `macros/omp-check.m4` as Jeroen says. On that line you test 
```
AS_IF([ test "x$with_openmp" != "xno" ]
```
but `with_openmp` should be `with_omp`. Because `$with_openmp` is undefined `"x$with_openmp"` is actually different from `"xno"` 100% of the time.

**Comment by ClementPernet** [comment]
Thanks for tracing back this bug.
This has been fixed in 735157564bb5e8a8eac9dbfc662008c7da64a04a
